module/data/actor/character.mjs

```diff
@@ -70,11 +73,15 @@ export default class CharacterData extends CreatureTemplate {
           min: new foundry.data.fields.NumberField({
             nullable: false, integer: true, min: 0, initial: 0, label: "DND5E.HitPointsMin"
           }),
-          max: new foundry.data.fields.NumberField({
-            nullable: false, integer: true, min: 0, initial: 0, label: "DND5E.HitPointsMax"
+          override: new foundry.data.fields.NumberField({
+            nullable: true, integer: true, min: 0, initial: null, label: "DND5E.HitPointsOverride"
           }),
           temp: new foundry.data.fields.NumberField({integer: true, initial: 0, min: 0, label: "DND5E.HitPointsTemp"}),
-          tempmax: new foundry.data.fields.NumberField({integer: true, initial: 0, label: "DND5E.HitPointsTempMax"})
+          tempmax: new foundry.data.fields.NumberField({integer: true, initial: 0, label: "DND5E.HitPointsTempMax"}),
+          bonuses: new foundry.data.fields.SchemaField({
+            level: new FormulaField({label: "DND5E.HitPointsBonusLevel"}),
```

Comment:
It does need to be deterministic because we don't have any place here to store a rolled value (that would have to be added to `HitPointsAdvancement`). I don't think I've seen anything that adds an extra rolled value per level, only flat values like Tough.